### PR TITLE
planner: add monitor for prepared plan cache memory usage 

### PR DESCRIPTION
**Pull request by fzzf678**
### What problem does this PR solve?

Issue Number: ref #37632

Problem Summary:

### What is changed and how it works?
Add  metrics to monitor prepared plan cache memory usage  and cached plan num  from the instance level
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```
After add panel:
LyNLxKi6p6

The eighth and ninth panel
planCache-memoryUsage-planNum-panels

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -134,6 +134,7 @@ func RegisterMetrics() {
 	prometheus.MustRegister(PanicCounter)
 	prometheus.MustRegister(PlanCacheCounter)
 	prometheus.MustRegister(PlanCacheMissCounter)
+	prometheus.MustRegister(InstancePlanCacheMemoryUsage)
```
Comment (qw4990) [inline metrics/metrics.go:137]: How about using `PlanCache` as the prefix: `PlanCacheInstanceMemoryUsage`?

Comment (fzzf678) [reply metrics/metrics.go]: updated

**Review by Reminiscent — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**